cypress/integration/tests/signUp.js

```diff
@@ -0,0 +1,104 @@
+      confirmPassword.equalToValidPassword);
+    SignUpPage.signOutIcon.should('be.visible');
+  });
+
+  it('TA-8.1: User unable to sign up with already registered email', () => {
+    SignUpPage.signUp(name.valid, email.registered,
+      password.valid, confirmPassword.equalToValidPassword);
+    SignUpPage.signUpError.should('contain', errorMessages.alreadyRegisteredEmail);
+  });
+
+  it('TA-9: User unable to sign up with missing Name', () => {
+    SignUpPage.enterSignUpEmail(email.unregistered);
+    SignUpPage.enterSignUpPassword(password.valid);
+    SignUpPage.enterSignUpConfirmPassword(confirmPassword.equalToValidPassword);
+    SignUpPage.submitSignUpForm();
+    SignUpPage.emptySignUpNameField.should('have.length', 1);
+    SignUpPage.emptySignUpNameField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-10: User unable to sign up with missing Email', () => {
+    SignUpPage.enterSignUpName(name.valid);
+    SignUpPage.enterSignUpPassword(password.valid);
+    SignUpPage.enterSignUpConfirmPassword(confirmPassword.equalToValidPassword);
+    SignUpPage.submitSignUpForm();
+    SignUpPage.emptyOrInvalidSignUpEmailField.should('have.length', 1);
+    SignUpPage.emptyOrInvalidSignUpEmailField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-11: User unable to sigh up with missing Password', () => {
+    SignUpPage.enterSignUpName(name.valid);
+    SignUpPage.enterSignUpEmail(email.unregistered);
+    SignUpPage.enterSignUpConfirmPassword(confirmPassword.equalToValidPassword);
+    SignUpPage.submitSignUpForm();
+    SignUpPage.emptyPasswordField.should('have.length', 1);
+    SignUpPage.emptyPasswordField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-12: User unable to sign up with missing Confirm Password', () => {
+    SignUpPage.enterSignUpName(name.valid);
+    SignUpPage.enterSignUpEmail(email.unregistered);
+    SignUpPage.enterSignUpPassword(password.valid);
+    SignUpPage.submitSignUpForm();
+    SignUpPage.emptyConfirmPasswordField.should('have.length', 1);
+    SignUpPage.emptyConfirmPasswordField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-13: User unable to sign up if Confirm Password is not matching Password', () => {
+    SignUpPage.signUp(name.valid, email.unregistered,
+      password.valid, confirmPassword.unequalToPassword);
+    cy.on('window:alert', (alertText) => {
+      expect(alertText).to.eq(errorMessages.passwordDoesntMatch);
+    });
+  });
+
+  it('TA-15: User unable to sign up with invalid Email', () => {
+    SignUpPage.signUp(name.valid, email.invalid,
+      password.valid, confirmPassword.equalToValidPassword);
```

Comment:
consider this syntax
```suggestion
    SignUpPage.signUp(
      name.valid,
      email.invalid,
      password.valid,
      confirmPassword.equalToValidPassword
    );
```
What you've done is really hard to catch